src/web/lib/hakuneko/frontend@classic-light/jobs.html

```diff
@@ -197,6 +275,29 @@
                 }
             }
 
+            /**
+             *
+             */
+            removeFromQueue(e) {
+                const index = e.model.index;
+                const job = this.jobList[index];
+                if (job) {
+                    Engine.DownloadManager.removeFromQueue(job);
+                    this.splice('jobList', index, 1);
```

Comment:
The method `DownloadManager.removeFromQueue()` can be called from anywhere in the software, so `this.jobList` may not reflect the correct entries.

Consider to introduce an event mechanism to update `this.jobList` whenever the queue in the `DownloadManager` is changed instead of setting it manually.